### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by DaliaMasilionyte**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,94 @@
...
+**Note:** We recommend you to lock your app to portrait orientation.
+
+---
+- **Important:** When using tools like Proguard to obfuscate your code, make sure to exclude Android Link SDK package (`com.argyle.*`) from the process, it may cause unexpected runtime issues otherwise. You can do this by adding this line to your `proguard-rules.pro:-keep class com.argyle. { *; }`
+  
+Our target configuration is currently set to the following:
+
+- `minSdkVersion = 23`
+- `compileSdkVersion = 30`
+- `targetSdkVersion = 30`
```
Comment (adrianestran) [inline README.md]: I think we can omit `targetSdk` and `compileSdk` in all future docs. Integrators only really care about `minSdk`.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,94 @@
...
+- `Android Support Library = 28.0.0`
```
Comment (adrianestran) [inline README.md]: We should also remove reference to support lib - we don't use this anymore.

File: UPGRADING.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+- To verify the change initiate a gradle sync and confirm no errors.
+
+- If you are using pro-guard, make sure you have the most up-to-date Proguard rules which can be found in our [integration guide](https://github.com/argyle-systems/argyle-link-android#readme).
+
+---
+# Migrating to Link 4
+
+If you're using a previous major version of Link and would like to migrate to the new version, please update the changed configuration parameters and then follow the usual [Android Link SDK version upgrade instruction](#upgrade-to-a-new-sdk-version) defined above.
```
Comment (adrianestran) [inline UPGRADING.md]: Should we call out which configuration params changed between 3 and 4? i.e pluginKey -> linkKey

File: README.md [hunk 4]
```diff
@@ -1,3 +1,94 @@
...
+- `Kotlin = 1.4.32`
+
+### 1. Adding the SDK dependency
```
Comment (adrianestran) [inline README.md]: Idea: Include a step before this with brief instructions on how to get access to `YOUR_LINK_KEY`. Something like

1. Access your Link API Key
- Log into your Console instance
- Navigate to the API Keys area under the Developer menu
- Copy your Sandbox or Production Link API Key for use in the next step

File: README.md [hunk 5]
```diff
@@ -1,3 +1,94 @@
...
+```
+dependencies {
+    implementation 'com.argyle:argyle-plugin-android-source:3.x.x'
+}
+```
+
+### 2. Creating the SDK configuration
+```kotlin
+val config = ArgyleConfig.Builder()
+    .loginWith("YOUR_LINK_KEY", "https://api-sandbox.argyle.com/v1", "USER_TOKEN")
```
Comment (adrianestran) [inline README.md]: I think we should definitely something in these readme's about what a `USER_TOKEN` is, and how integrators should use them

**Review by adrianestran — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (Varnius) [reply README.md]: We could link to the [user token guide](https://argyle.com/docs/products/returning-users-experience) we have in our docs

File: UPGRADING.md (hunk 3 above)
Comment (DaliaMasilionyte) [reply UPGRADING.md]: @adrianestran I had initially added this and then removed it, as `pluginKey` is an unnamed parameter - therefore nothing changes on client's side. Should I add it back?

File: UPGRADING.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+# Android SDK version upgrade guide
+
+## Learn how to upgrade to the newest Link version.
+If you are looking to integrate Argyle Link for the first time navigate to [integration guide](https://github.com/argyle-systems/argyle-link-android#readme).
```
Comment (adrianestran) [inline UPGRADING.md:4]: Should we be linking to `https://argyle.com/docs/products/argyle-link#android-sdk-integration`?

Comment (DaliaMasilionyte) [reply UPGRADING.md:4]: We’ve added integration details to readme not sure why we wouldn’t use it?

Comment (adrianestran) [reply UPGRADING.md:4]: Oh sorry, I'm being dumb - thats the change we're merging 🤦‍♂️

**Review by adrianestran — APPROVED** [review]

**Pull request merged**